source/MaterialXGenGlsl/Nodes/UnlitSurfaceNodeGlsl.cpp

```diff
@@ -0,0 +1,56 @@
+//
+// TM & (c) 2022 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenGlsl/Nodes/UnlitSurfaceNodeGlsl.h>
+
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/GenContext.h>
+
+MATERIALX_NAMESPACE_BEGIN
+
+ShaderNodeImplPtr UnlitSurfaceNodeGlsl::create()
+{
+    return std::make_shared<UnlitSurfaceNodeGlsl>();
+}
+
+void UnlitSurfaceNodeGlsl::emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const
+{
+    const GlslShaderGenerator& shadergen = static_cast<const GlslShaderGenerator&>(context.getShaderGenerator());
+
+    BEGIN_SHADER_STAGE(stage, Stage::PIXEL)
+
+        // Declare the output variable
+        const ShaderOutput* output = node.getOutput();
+        shadergen.emitLineBegin(stage);
+        shadergen.emitOutput(output, true, true, context, stage);
+        shadergen.emitLineEnd(stage);
+
+        const string outColor = output->getVariable() + ".color";
+        const string outTransparency = output->getVariable() + ".transparency";
+
+        const ShaderInput* emission = node.getInput("emission");
+        const ShaderInput* emissionColor = node.getInput("emission_color");
+        shadergen.emitLine(outColor + " = " + shadergen.getUpstreamResult(emission, context) + " * " + shadergen.getUpstreamResult(emissionColor, context), stage);
+        
+        if (context.getOptions().hwTransparency)
+        {
+            const ShaderInput* transmission = node.getInput("transmission");
+            const ShaderInput* transmissionColor = node.getInput("transmission_color");
+            shadergen.emitLine(outTransparency + " = " + shadergen.getUpstreamResult(transmission, context) + " * " + shadergen.getUpstreamResult(transmissionColor, context), stage);
+
+            const ShaderInput* opacity = node.getInput("opacity");
+            const string surfaceOpacity = shadergen.getUpstreamResult(opacity, context);
+            shadergen.emitLine(outColor + " *= " + surfaceOpacity, stage);
```

Comment:
Checking here that this doesn't quite match the osl implementation in that it does not mix transmission here?

Reply:
Yes, my thinking here is that the HW alpha blending would already do that for us. But just let me know if that's not the case. Again this logic is taken from the PBR version of the "surface" node, so if a change is needed it might be best to do as part of a deeper look at all transparency handling, so it's consistent for all surface shaders.

Reply:
I see so alpha blending gives you some "cheap" way to fake transmission. Let's add a note for this, but from past implementations of h/w transmission for rasterizers I kind of remember including transmission w/o reliance on alpha blending.